src/main/java/randoop/generation/GrtImpurity.java

```diff
@@ -0,0 +1,513 @@
+ * branches, improving coverage for the program under test.
+ *
+ * <p>[TODO: As described in the GRT paper, "GRT Impurity" also generates more effective test with
+ * shorter length by reducing the number of redundant sequences that do not side-effect the state of
+ * an object. That is not implemented here.]
+ *
+ * <p>For details about fuzzing, see {@link #fuzz}.
+ */
+@SuppressWarnings("NotJavadoc") // perhaps https://github.com/google/error-prone/issues/3904
+public class GrtImpurity {
+
+  /** Ways to fuzz a string. */
+  private enum StringFuzzingOperation {
+    /** Insert a random character at a random index in the string. */
+    INSERT,
+
+    /** Remove a character at a random index in the string. */
+    REMOVE,
+
+    /** Replace a character at a random index in the string with a random character. */
+    REPLACE,
+
+    /** Take a random substring of the string. */
+    SUBSTRING
+  }
+
+  /** The standard deviation of the Gaussian distribution used to generate fuzzed numbers. */
+  private static final double GAUSSIAN_STD = GenInputsAbstract.grt_impurity_stddev;
+
+  /** Do not instantiate. */
+  private GrtImpurity() {
+    throw new Error("Do not instantiate");
+  }
+
+  /**
+   * Generate an extended fuzzed sequence for the given sequence.
+   *
+   * <p>This is the main entry point for GRT Impurity. This method appends additional fuzzing
+   * statements to the input sequence to fuzz the input object. This will generate more object
+   * states and potentially improve coverage for the unit test generated by Randoop by triggering
+   * more branches.
+   *
+   * <p>Different types have different fuzzing strategies:
+   *
+   * <ul>
+   *   <li><b>Numbers (int, short, long, float, double, and their wrapper classes):</b> Fuzzed using
+   *       a Gaussian distribution with the original value as the mean (mu). The fuzzed value is
+   *       generated as mu + N(0, GAUSSIAN_STD). For short values, if the fuzzed value exceeds the
+   *       short range, it is truncated.
+   *   <li><b>Strings:</b> A random fuzzing operation is selected from the {@link
+   *       StringFuzzingOperation} enum.
+   *   <li><b>Other Objects:</b> [TODO: Further implementation required] Methods are analyzed for
+   *       side-effects to enhance test effectiveness by focusing on these interactions.
+   * </ul>
+   *
+   * <p>This method does not fuzz void, char, boolean, or byte types.
+   *
+   * @param sequence the (non-null) sequence to fuzz
+   * @return a sequence consisting of {@code sequence} with additional fuzzing statements appended
+   *     at the end, and a count of the number of fuzzing statements added. If no fuzzing statements
+   *     are added, the original sequence is returned, and the count is 0.
+   */
+  public static GrtImpurityAndNumStatements fuzz(Sequence sequence) {
+    // The number of fuzzing statements added to the sequence.
+    FuzzStatementOffset fuzzStatementOffset = new FuzzStatementOffset();
```

Comment:
The documentation says it's a number, but the variable name uses "offset".  This is inconsistent.